tests/profiling_v2/collector/test_memalloc.py

```diff
@@ -372,3 +419,434 @@ def lotsa_allocs(ev):
+        )
+
+        assert abs(rel_size - actual_rel_size) < 0.10
+        assert abs(rel_count - actual_rel_count) < 0.15
+
+    print(f"Successfully validated allocation sampling accuracy for sample_interval={sample_interval}")
+    print(f"Captured {len(allocation_samples)} allocation samples representing {total_allocation_count} allocations")
+
+
+def test_memory_collector_allocation_tracking_across_snapshots():
+    mc = memalloc.MemoryCollector(heap_sample_size=64)
+
+    with mc:
+        initial_samples = mc.test_snapshot()
+
+        assert all(
+            sample.alloc_size > 0 for sample in initial_samples
+        ), "Initial snapshot should have alloc_size>0 (new allocations)"
+
+        data_to_free = []
+        for i in range(10):
+            data_to_free.append(one(256))
+
+        data_to_keep = []
+        for i in range(10):
+            data_to_keep.append(two(512))
+
+        del data_to_free
+
+        second_samples = mc.test_snapshot()
+
+        freed_samples = [s for s in second_samples if s.in_use_size == 0]
+        live_samples = [s for s in second_samples if s.in_use_size > 0]
+
+        assert len(freed_samples) > 0, "Should have some freed samples after deletion"
+
+        assert len(live_samples) > 0, "Should have some live samples"
+
+        assert all(sample.in_use_size == 0 for sample in freed_samples), "Freed samples should have in_use_size=0"
+        assert all(sample.in_use_size > 0 for sample in live_samples), "Live samples should have in_use_size>0"
+
+        for sample in second_samples:
+            assert isinstance(sample.size, int) and sample.size > 0, f"Invalid size: {sample.size}"
+            assert isinstance(sample.count, int) and sample.count > 0, f"Invalid count: {sample.count}"
+            assert (
+                isinstance(sample.in_use_size, int) and sample.in_use_size >= 0
+            ), f"Invalid in_use_size: {sample.in_use_size}"
+            assert (
+                isinstance(sample.alloc_size, int) and sample.alloc_size >= 0
+            ), f"Invalid alloc_size: {sample.alloc_size}"
+
+        one_freed_samples = [
+            sample
+            for sample in second_samples
+            if has_function_in_traceback(sample.frames, "one") and sample.in_use_size == 0
+        ]
+
+        assert len(one_freed_samples) > 0, "Should have freed samples from function 'one'"
+        assert all(sample.in_use_size == 0 and sample.alloc_size > 0 for sample in one_freed_samples)
+
+        two_live_samples = [
+            sample
+            for sample in second_samples
+            if has_function_in_traceback(sample.frames, "two") and sample.in_use_size > 0
+        ]
+
+        assert len(two_live_samples) > 0, "Should have live samples from function 'two'"
+        assert all(sample.in_use_size > 0 and sample.alloc_size > 0 for sample in two_live_samples)
+
+        del data_to_keep
+
+
+def test_memory_collector_python_interface_with_allocation_tracking():
```

Comment:
Unless I'm missing something, this seems to be testing the same thing as the previous test?